src/main/java/com/onarandombox/MultiverseCore/api/action/ActionResponse.java

```diff
@@ -0,0 +1,107 @@
+package com.onarandombox.MultiverseCore.api.action;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A response that have multiple results.
+ */
+public class ActionResponse implements ActionResult {
+    public static ActionResponse create() {
+        return new ActionResponse();
+    }
+
+    public static ActionResponse create(boolean continueIfFail) {
+        return new ActionResponse(continueIfFail);
+    }
+
+    public static ActionResponse of(ActionResult result) {
+        return new ActionResponse().addResult(result);
+    }
+
+    private final Set<ActionResult> results;
+    private final boolean continueIfFail;
+
+    private boolean isSuccessful = true;
+
+    public ActionResponse() {
+        this(false);
+    }
+
+    /**
+     * @param continueIfFail If true, the response will continue to add results even if one of the results is not successful.
+     */
+    public ActionResponse(boolean continueIfFail) {
+        this.results = new HashSet<>();
+        this.continueIfFail = continueIfFail;
+    }
```

Comment:
Does this have to be public? We should prefer private constructors in general.

Reply:
Yea protected makes sense